core/metric.py

```diff
@@ -24,24 +26,46 @@ def __init__(self, config):
         """ """
         pass
 
-    @abc.abstractmethod
-    def clear(self, scope, params):
-        """
-        clear current value
-        Args:
-            scope: value container
-            params: extend varilable for clear
-        """
-        pass
+    def clear(self, scope=None):
+        """ """
+        if scope is None:
+            scope = fluid.global_scope()
 
-    @abc.abstractmethod
-    def calculate(self, scope, params):
-        """
-        calculate result
-        Args:
-            scope: value container
-            params: extend varilable for clear
-        """
+        place = fluid.CPUPlace()
+        for key in self._global_communicate_var:
```

Comment:
what's the meaning of `_global_communicate_var ` ?

Reply:
variables that need to be communicated between trainers， used to calculate global metrics